### ISSUE TITLE
[Egg Request]: PowerNukkitX

### ISSUE THREAD

**Issue by rac14** [parkervcp/eggs#2152]
### Does this expand an already existing service

Yes

### Link to a game or software

https://github.com/PowerNukkitX/PowerNukkitX

### Links for the download

https://github.com/PowerNukkitX/PNX-CLI/releases

### Links for the install docs

https://doc.powernukkitx.cn/en-us/Get_Started.html

**Comment by QuintenQVD0** [comment]
you prefer that is install over the CLI or directly with the .jar and libs?

**Comment by ProtectorYT364** [comment]
> you prefer that is install over the CLI or directly with the .jar and libs?

with the .jar and libs

**Comment by QuintenQVD0** [comment]
@ProtectorYT364 

please fully test:
(you will have to type the lang you want on first startup in the console)
```json
{
    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PTERODACTYL PANEL - PTERODACTYL.IO",
    "meta": {
        "version": "PTDL_v2",
        "update_url": null
    },
    "exported_at": "2023-03-05T09:48:53+01:00",
    "name": "PowerNukkitX",
    "author": "josdekurk@gmail.com",
    "description": "Make Nukkit Great Again!",
    "features": null,
    "docker_images": {
        "ghcr.io\/parkervcp\/yolks:java_17": "ghcr.io\/parkervcp\/yolks:java_17"
    },
    "file_denylist": [],
    "startup": "java -Xms128M -XX:MaxRAMPercentage=95.0 -Dfile.encoding=UTF-8 -Djansi.passthrough=true -Dterminal.ansi=true -XX:+UnlockExperimentalVMOptions -XX:+UseStringDeduplication -XX:+EnableJVMCI  --module-path=.\/libs\/truffle-api-22.2.0.jar:.\/libs\/graal-sdk-22.2.0.jar: --add-opens java.base\/java.lang=ALL-UNNAMED --add-opens java.base\/java.io=ALL-UNNAMED -cp .\/powernukkitx.jar:.\/libs\/* cn.nukkit.Nukkit",
    "config": {
        "files": "{\r\n    \"server.properties\": {\r\n        \"parser\": \"properties\",\r\n        \"find\": {\r\n            \"server-ip\": \"0.0.0.0\",\r\n            \"server-port\": \"{{server.build.default.port}}\",\r\n            \"query.port\": \"{{server.build.default.port}}\"\r\n        }\r\n    }\r\n}",
        "startup": "{\r\n    \"done\": \")! For help, type \"\r\n}",
        "logs": "{}",
        "stop": "stop"
    },
    "scripts": {
        "installation": {
            "script": "#!\/bin\/bash\r\n\r\napt update\r\napt -y install curl git jq tar\r\n\r\n## get release info and download links\r\nLATEST_JSON=$(curl --silent \"https:\/\/api.github.com\/repos\/PowerNukkitX\/PowerNukkitX\/releases\/latest\")\r\nRELEASES=$(curl --silent \"https:\/\/api.github.com\/repos\/PowerNukkitX\/PowerNukkitX\/releases\")\r\nJAR=powernukkitx.jar\r\nLIBS=libs.tar.gz\r\n\r\nif [ -z \"${VERSION}\" ] || [ \"${VERSION}\" == \"latest\" ]; then\r\n    DOWNLOAD_JAR_URL=$(echo ${LATEST_JSON} | jq .assets | jq -r .[].browser_download_url | grep -i ${JAR} | head -1)\r\n    DOWNLOAD_LIBS_URL=$(echo ${LATEST_JSON} | jq .assets | jq -r .[].browser_download_url | grep -i ${LIBS} | head -1)\r\nelse\r\n    VERSION_CHECK=$(echo ${RELEASES} | jq -r --arg VERSION \"${VERSION}\" '.[] | select(.tag_name==$VERSION) | .tag_name')\r\n    if [ \"${VERSION}\" == \"${VERSION_CHECK}\" ]; then\r\n        DOWNLOAD_JAR_URL=$(echo ${RELEASES} | jq -r --arg VERSION \"${VERSION}\" '.[] | select(.tag_name==$VERSION) | .assets[].browser_download_url' | grep -i ${JAR} | head -1)\r\n        DOWNLOAD_LIBS_URL=$(echo ${RELEASES} | jq -r --arg VERSION \"${VERSION}\" '.[] | select(.tag_name==$VERSION) | .assets[].browser_download_url' | grep -i ${LIBS} | head -1)\r\n    else\r\n        echo -e \"defaulting to latest release\"\r\n        DOWNLOAD_JAR_URL=$(echo ${LATEST_JSON} | jq .assets | jq -r .[].browser_download_url | grep -i ${JAR} | head -1)\r\n        DOWNLOAD_LIBS_URL=$(echo ${LATEST_JSON} | jq .assets | jq -r .[].browser_download_url | grep -i ${LIBS} | head -1)\r\n    fi\r\nfi\r\n\r\nmkdir -p \/mnt\/server\r\ncd \/mnt\/server\r\nrm powernukkitx.jar.old\r\n\r\nif [ -f powernukkitx.jar ]; then\r\n\tmv powernukkitx.jar powernukkitx.jar.old\r\nfi\r\n\r\necho -e \"\\nRunning: curl -sSL -o powernukkitx.jar ${DOWNLOAD_JAR_URL}\"\r\ncurl -sSL -o powernukkitx.jar ${DOWNLOAD_JAR_URL}\r\n\r\necho -e \"\\nRunning: curl -sSL -o libs.tar.gz ${DOWNLOAD_LIBS_URL}\"\r\ncurl -sSL -o libs.tar.gz ${DOWNLOAD_LIBS_URL}\r\n\r\necho -e \"\\nUnpacking the libs\"\r\ntar xvf libs.tar.gz --strip-components=1\r\nrm libs.tar.gz\r\n\r\nif [ ! -f server.properties ]; then\r\n    echo -e \"Downloading MC server.properties\"\r\n    curl -o server.properties https:\/\/raw.githubusercontent.com\/parkervcp\/eggs\/master\/minecraft\/java\/server.properties\r\nfi\r\n## install end\r\necho \"-----------------------------------------\"\r\necho \"Installation completed...\"\r\necho \"-----------------------------------------\"",
            "container": "debian:bullseye-slim",
            "entrypoint": "bash"
        }
    },
    "variables": [
        {
            "name": "Version",
            "description": "",
            "env_variable": "VERSION",
            "default_value": "latest",
            "user_viewable": true,
            "user_editable": true,
            "rules": "required|string|max:20",
            "field_type": "text"
        }
    ]
}
```

**Comment by QuintenQVD0** [comment]
Closed as of #2354
